solvation_analysis/solvation.py

```diff
@@ -6,29 +6,142 @@
+
+    Returns
+    -------
+        AtomGroup
+
+    """
+    assert isinstance(selection, (mda.core.groups.Residue,
+                                  mda.core.groups.ResidueGroup,
+                                  mda.core.groups.Atom,
+                                  mda.core.groups.AtomGroup)), \
+        "central_species must be one of the preceding types"
+    if isinstance(selection, (mda.core.groups.Residue, mda.core.groups.ResidueGroup)):
+        selection = selection.atoms
+    if isinstance(selection, mda.core.groups.Atom):
+        selection = u.select_atoms(f'index {selection.index}')
+    return selection
+
+
+# test this
+def get_n_shells(u, central_species, n_shell=2, radius=3, ignore_atoms=None):
+    """
+    A list containing the nth shell at the nth index. Note that the shells have 0 intersection.
+    For example, calling get_n_shells with n_shell = 2 would return: [central_species, first_shell, second_shell].
+    This scales factorially so probably don't go over n_shell = 3
+
+    Parameters
+    ----------
+        u : Universe
+            universe that contains central species
+        central_species : Atom, AtomGroup, Residue, or ResidueGroup
+        n_shell : int
+            number of shells to return
+        radius : float or int
+            radius used to select atoms in next shell
+        ignore_atoms : AtomGroup
+            these atoms will be ignored
+
+    Returns
+    -------
+        List[AtomGroups] :
+            List of n shells
+
     """
-    Placeholder function to show example docstring (NumPy format)
+    if n_shell > 3:
+        warnings.warn('get_n_shells scales factorially, very slow')
+    central_species = get_atom_group(u, central_species)
+    if not ignore_atoms:
+        ignore_atoms = u.select_atoms('')
+
 
-    Replace this function and doc string for your own project
+def get_closest_n_mol(u, central_species, n_mol=5, radius=3):
+    """
+    Returns the closest n molecules to the central species, an array of their resids,
+    and an array of the distance of the closest atom in each molecule.
 
     Parameters
     ----------
-    with_attribution : bool, Optional, default: True
-        Set whether or not to display who the quote is from
+        u : Universe
+            universe that contains central species
+        central_species : Atom, AtomGroup, Residue, or ResidueGroup
+        n_mol : int
+            The number of molecules to return
+        radius : float or int
+            an initial search radius to look for closest n mol
 
     Returns
     -------
-    quote : str
-        Compiled string including quote and optional attribution
+        AtomGroup (molecules), np.Array (resids), np.Array (distances)
+
     """
+    central_species = get_atom_group(u, central_species)
+    coords = central_species.center_of_mass()
+    str_coords = " ".join(str(coord) for coord in coords)
+    partial_shell = u.select_atoms(f'point {str_coords} {radius}')
```

Comment:
AFAIK this calls `capped_distances` under the hood. 
(https://github.com/MDAnalysis/mdanalysis/blob/a6d4ffc26375a44d4aaa53de88d193b9e1e645e9/package/MDAnalysis/lib/distances.py#L319).
It may be more efficient to use it directly, and it also returns the distances as an optional return argument.